ff2mpv.py

```diff
@@ -30,13 +34,58 @@ def main():
         path = os.environ.get("PATH")
         os.environ["PATH"] = f"/usr/local/bin:{path}"
 
+    if "cookies" in message and is_whitelisted(url, get_whitelist()):
+        cookies_fname = create_cookiefile(message.get("cookies"));
+        ytdloptions["cookies"] = cookies_fname
+        additional_mpv_args += ['--cookies', '--cookies-file={}'.format(cookies_fname)]
+
+    mpv_ytdloptions = '--ytdl-raw-options-append={}'.format(
+        ",".join("{}={}".format(k,v) for k,v in ytdloptions.items()))
+
+    args = ['mpv', '--no-terminal', mpv_ytdloptions] + additional_mpv_args + ["--", url]
+
     subprocess.Popen(args, **kwargs)
 
     # Need to respond something to avoid "Error: An unexpected error occurred"
     # in Browser Console.
     send_message("ok")
 
 
+def create_cookiefile(cookies):
```

Comment:
This can probably be cleaned up a little, but I'll take a pass at it once the other comments are addressed.

Reply:
I took `create_cookiefile()` the completely from @boi4. Maybe he might be able to help with that...

Reply:
Yes, I agree that this isn't the most beautiful and failsafe function (catching KeyErrors would make sense). The format that mpv and youtube-dl need is the netscape cookie format. Here is some more information: https://curl.se/docs/http-cookies.html

Reply:
Yeah, don't worry too much about cleaning this up. I'll take care of it with my changes tonight.